### PR TITLE
feat: Add the --valueFiles flag

### PR DESCRIPTION
**Pull request by edwardstudy**

### CODE REVIEW

**Comment by edwardstudy** [comment]
Passed testing:

```
cat /Volumes/journey/workspace/bcf-deployer/config/cf_components_sizing.yaml
---
sizing:
  diego_cell:
    count: 4

IBMMasterMBP:bcf-deployer bjxzi$ /Volumes/journey/workspace/bcf-deployer/scripts/helm_mac update-config cf.20190131.134059 -f /Volumes/journey/workspace/bcf-deployer/config/cf_components_sizing.yaml --set-value=kube.control_plane_size=2,kube.diego_cell_size=4
Info: update successfully
IBMMasterMBP:bcf-deployer bjxzi$ diff <(helm get values cf.20190131.134059  --all --revision 6) <(helm get values cf.20190131.134059  --all) -u
--- /dev/fd/63	2019-02-19 16:04:57.000000000 +0800
+++ /dev/fd/62	2019-02-19 16:04:57.000000000 +0800
@@ -224,7 +224,7 @@
   WABDAV_PUBLIC_ENDPOINT: https://blobstore.((DOMAIN))
 kube:
   auth: rbac
   control_plane_size: 2
   diego_cell_partition: 0
   diego_cell_size: 2
+  diego_cell_size: 4
   external_ips:
@@ -514,7 +514,7 @@
       request: 99
   diego_cell:
     affinity: {}
-    count: 2
+    count: 4
     cpu:
       limit: null
       request: 4000
```

File: main.go [hunk 1]
```diff
@@ -2,28 +2,51 @@ package main
...
 	"github.com/spf13/cobra"
 	yaml "gopkg.in/yaml.v1"
 	"k8s.io/helm/pkg/helm"
 	"k8s.io/helm/pkg/strvals"
 )
 
+type cmdFlags struct {
+	cliValues       []string
+	resetValues     bool
+	templateNumbers []string
```
Comment (zhanggbj) [inline main.go]: Hi @edwardstudy 
`templateNumbers ` is no longer used , this can be removed. Thanks!

File: main.go [hunk 2]
```diff
@@ -75,23 +103,23 @@ func (cmd *updateConfigCommand) run() error {
 		opt = helm.ReuseValues(true)
 	}
 
-	newValues, _ := yaml.Marshal(dest)
 	_, err = cmd.client.UpdateReleaseFromChart(
 		ls.Releases[0].Name,
 		ls.Releases[0].Chart,
-		helm.UpdateValueOverrides(newValues),
+		helm.UpdateValueOverrides(rawVals),
```
Comment (zhanggbj) [inline main.go]: And would you please help confirm, after the update, you can still get the full set of the values. It seems that the input of helm.UpdateValueOverrides is the values you need to update, the final result will only have these values, and lose the original ones.

**Review by zhanggbj — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (edwardstudy) [reply main.go]: I just fetched values thou `helm get values cf.20190131.134059  --all`  and got all values including what I didn't update.

File: main.go [hunk 3]
```diff
@@ -64,9 +89,22 @@ func (cmd *updateConfigCommand) run() error {
...
-	var base map[interface{}]interface{}
-	yaml.Unmarshal([]byte(ls.Releases[0].Config.Raw), &base)
-	dest := mergeValues(base, cmd.values)
+	var preVals map[string]interface{}
+	err = yaml.Unmarshal([]byte(ls.Releases[0].Config.Raw), &preVals)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to unmarshal raw values: %v", ls.Releases[0].Config.Raw)
+	}
+
+	updatedVals, err := generateUpdatedValues(cmd.valueFiles, cmd.values)
```
Comment (zhanggbj) [inline main.go]: @edwardstudy 
I think the updatedVals is a little bit confusing, how about `preferredVals` which means which you would like to change to?

File: main.go [hunk 4]
```diff
@@ -64,9 +89,22 @@ func (cmd *updateConfigCommand) run() error {
 		return err
 	}
 
-	var base map[interface{}]interface{}
-	yaml.Unmarshal([]byte(ls.Releases[0].Config.Raw), &base)
-	dest := mergeValues(base, cmd.values)
+	var preVals map[string]interface{}
```
Comment (zhanggbj) [inline main.go:92]: the `ls.Releases[0].Config.Raw` is not just Vals, including values and other configs, right? how about we name it back to `base`?

File: main.go [hunk 5]
```diff
@@ -75,47 +113,94 @@ func (cmd *updateConfigCommand) run() error {
...
 		}
-		// If they are bosh map, merge them
-		dest[k] = mergeValues(destMap, nextMap)
+
+		// Convert destMap key as string
+		convertedDestMap := map[string]interface{}{}
+		for k,v := range destMap{
+			convertedDestMap[k.(string)] = v
+		}
+
```
Comment (zhanggbj) [inline main.go]: Line 148-152 and Line 162-165 are basically the same, could you please change to use a function?

**Review by zhanggbj — CHANGES_REQUESTED** [review]

File: main.go (hunk 3 above)
Comment (edwardstudy) [reply main.go]: OK

**Pull request merged**